### ISSUE TITLE
Media Query Directive

### ISSUE THREAD

**Issue by kyleledbetter** [Teradata/covalent#20]
A directive that conditionally applies:

- [x] CSS Classes
- [x] Other directives
- [x] Attributes
- [x] Also use within TypeScript on a Component/Pipe/Class/etc

based on media-query-like min/max width/height

---

Something like:
`<a md-list-item [routerLink]="[item.route]” mediaQuery(‘max-width:600px, (click)="list.toggle()")>`
> where the `(click)="list.toggle()"` only gets applied if `max-width:600px`

Hoping NgZone & MediaQueryList can help:

```
@Component({
  selector: 'my-app',
  styles: [`
    .box { width: 300px; height: 300px; }
  `],
  template : `<div class="box" [style.backgroundColor]="color"></div>
  `,
  directives : []
})
export class App {  
  public color: string = 'black';

  constructor(zone: NgZone) {
    const mql: MediaQueryList = window.matchMedia('(min-width: 600px)');

    mql.addListener((mql: MediaQueryList) => { 
      zone.run( () => { // Change the property within the zone, CD will run after
        this.color = mql.matches ? 'blue' : 'yellow';
      });
    });
  }
}
```

**Comment by emoralesb05** [comment]
Cant really add directives on the fly since they are added and compiled before the app starts.

**Comment by kyleledbetter** [comment]
closed with #22
